sklearn/utils/isfinite.pyx

```diff
@@ -0,0 +1,65 @@
+# Author: jakirkham, Meekail Zain, Thomas Fan
+
+from libc.math cimport isnan as c_isnan, isinf as c_isinf
+
+cimport cython
+cimport numpy as cnp
+import numpy as np
+cnp.import_array()
+
+
+cdef fused fprecision:
+    cnp.npy_float32
+    cnp.npy_float64
+
+cdef fused const_fprecision:
+    const cnp.npy_float32
+    const cnp.npy_float64
+
+def cy_isfinite(cnp.ndarray a, bint allow_nan=False):
+    cdef:
+        char* a_data
+        cnp.NPY_TYPES a_type
+        Py_ssize_t a_size = a.size
+        bint disallow_nan
+        int result
+        bint err
+
+    if a.dtype not in {np.float32, np.float64}:
+        raise TypeError("Unsupported array type: %s" % repr(a.dtype))
+
+    with nogil:
+        a_data = a.data
+        a_type = <cnp.NPY_TYPES>a.descr.type_num
+
+        disallow_nan = not allow_nan
```

Comment:
I do not see the need for negating `allow_nan`. Can we pass `allow_nan` directly into  `_isfinite`?